spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.spark.JobGroupInfo;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Computes the statistic of the given columns and stores it as Puffin files. */
+public class AnalyzeTableSparkAction extends BaseSparkAction<AnalyzeTableSparkAction>
+    implements AnalyzeTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+  private final Table table;
+  private Set<String> columnsToBeAnalyzed = ImmutableSet.of();
+  private Set<String> statsToBecollected =
+      Sets.newHashSet(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1);
+  private Long snapshotId;
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    String desc = String.format("Analyzing table %s", table.name());
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    LOG.info("Starting the analysis of {}", table.name());
+    List<AnalysisResult> analysisResults =
+        statsToBecollected.stream()
+            .map(
+                statsName -> {
+                  switch (statsName) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      return generateNDVAndCommit();
+                    default:
+                      return ImmutableAnalyzeTable.AnalysisResult.builder()
+                          .statsName(statsName)
+                          .statsCollected(false)
+                          .addAllErrors(Lists.newArrayList("Stats type not supported"))
+                          .build();
+                  }
+                })
+            .collect(Collectors.toList());
+    return ImmutableAnalyzeTable.Result.builder().analysisResults(analysisResults).build();
+  }
+
+  private boolean analyzeableTypes(Set<String> columns) {
+    return columns.stream()
+        .anyMatch(
+            columnName -> {
+              Types.NestedField field = table.schema().findField(columnName);
+              if (field == null) {
+                throw new ValidationException("No column with %s name in the table", columnName);
+              }
+              Type type = field.type();
+              return type.isListType()
+                  || type.isMapType()
+                  || type.isNestedType()
+                  || type.isStructType();
+            });
+  }
+
+  private AnalysisResult generateNDVAndCommit() {
+    try {
+
+      if (snapshotId == null) {
+        snapshotId = table.currentSnapshot().snapshotId();
```

Comment:
Hm, I was initially thinking it will be cleaner to make it static default the constructor itself like:
```
  AnalyzeTableSparkAction(SparkSession spark, Table table) {
    super(spark);
    this.table = table;
    this.snapshotId = table.currentSnapshot().snapshotId();
  }
  ```

Are you thinking to make it dynamic where the user calls table.refresh() between creating the action and execute?  But I see a problem, if you call execute again, it wont update as snapshotId is set already.

I think what bothers me is that it non-determisticallly sets the state by calling execute(), and also in generateNDVAndCommit().  How about making the snapshot a Long and keeping it null.  Then just have a method 

  private long snapshotId() {
    return snapshotId != null ? snapshotId : table.currentSnapshot().snapshotId();
  }

that we use everywhere.  This is inspired from DataTableScan::snapshot()